sunpy/map/mapbase.py

```diff
@@ -1092,26 +1099,21 @@ def world_to_pixel(self, coordinate, origin=0):
         y : `~astropy.units.Quantity`
             Pixel coordinate on the CTYPE2 axis.
         """
-        if not isinstance(coordinate, (SkyCoord,
-                                       astropy.coordinates.BaseCoordinateFrame)):
-            raise ValueError(
-                "world_to_pixel takes a Astropy coordinate frame or SkyCoord instance.")
-
-        native_frame = coordinate.transform_to(self.coordinate_frame)
-        lon, lat = u.Quantity(self._get_lon_lat(native_frame)).to(u.deg)
-        x, y = self.wcs.wcs_world2pix(lon, lat, origin)
+        self._check_origin(origin)
+        x, y = self.wcs.world_to_pixel(coordinate)
+        if origin == 1:
+            x += 1
+            y += 1
 
         return PixelPair(x * u.pixel, y * u.pixel)
 
     @u.quantity_input
-    def pixel_to_world(self, x: u.pixel, y: u.pixel, origin=0):
+    def pixel_to_world(self, x: u.pixel, y: u.pixel, origin=None):
```

Comment:
```suggestion
    def pixel_to_world(self, x: u.pixel, y: u.pixel, origin=0):
```

Reply:
Woops, I forgot to leave that out, thanks for catching.